### PR TITLE
[slack] Basic bot reacting to events and .gitignore update

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

Basic bot greeting users("Hello X") when user X says "Hello Hue" and .gitignore updated for .env file.

## How was this patch tested?

- Manually tested

A quick note: 
- Basic Flask app created for web server responding to events
- ngrok required for public url forwarding request to localhost

Update:
- If using ngrok (free version), then public url changes after every session expiration or closing the terminal, therefore we need to update the url in "Events Subscription" in our Slack app to continue to monitor events.
- Also need to make a .env file for all tokens/secrets since it is not tracked by git.

### CODE REVIEW

File: tools/slack/bot.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import os
+from slack_sdk import WebClient
+from pathlib import Path
+from dotenv import load_dotenv
+from flask import Flask
+from slackeventsapi import SlackEventAdapter
+
+#Create .env file and load it
+load_dotenv()
+app = Flask(__name__)
```
Comment (romainr) [inline tools/slack/bot.py:10]: Add in a doc ways to have the webserver with pros/cons?
Flask, Hue url, Hue standalone command?

File: tools/slack/bot.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  """Greets the user if user says hello hue, 
+  else sends the same message as reply """
+  event = payload.get('event', {}) # Get event else return empty dictionary
+  channel_id = event.get('channel')
+  user_id = event.get('user')
+  text = event.get('text')
+ 
+  if BOT_ID != user_id:
+    if "hello hue" in text.lower():
+      text = f'Hello <@{user_id}> :wave:'
```
Comment (romainr) [inline tools/slack/bot.py:27]: Nice!

In next iteration the goal could be to reply:

Hello Harsh! You have X SQL queries.

where X is the just the number of queries of the user. Check Document2 model in desktop and how it is used lib/notebook and https://docs.gethue.com/developer/api/#python-1

File: tools/slack/bot.py (hunk 1 above)
Comment (Harshg999) [reply tools/slack/bot.py:10]: I am not able to get what is meant by "Hue url", so like we can have something like "demo.gethue.com/slack/events" or "demo.gethue.com/slack/actions" ?

And "Hue standalone command" ? More like a "Slash command" in slack ?

Comment (romainr) [reply tools/slack/bot.py:10]: The webserver could be: Flask, directly part of Hue, running outside of Hue. 
I put an example of Hue/Django command showing how from the Hue code base a server can be started on the side.

**Comment by Harshg999** [comment]
Closing this PR as it has become outdated now.

**Pull request closed without merging**